[WFLY-19954] A follow up to add tests on the vertx instance usage in opentelemetry subsystem

gaol opened the pull request:
This is a follow up to add tests on the vertx instance usage in opentelemetry subsystem

Related issue is: https://issues.redhat.com/browse/WFLY-19954

If a new WFLY issue is a better, I can create one to update.

rhusar commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/preview/microprofile/telemetry/AbstractOpenTelemetryIntegrationTest.java:
```diff
@@ -0,0 +1,74 @@
+    static WebArchive createDeployment(String deploymentName) {
+        return ShrinkWrap
+                .create(WebArchive.class, deploymentName + ".war")
+                .addPackage(AbstractOpenTelemetryIntegrationTest.class.getPackage())
```
All the tests extending the AbstractOpenTelemetryIntegrationTest are run as client, so do not pollute the deployment with the test class itself.

gaol replied:
Thank you @rhusar , updated to only package the REST resources in the deployment.

bstansberry commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/preview/microprofile/telemetry/AbstractOpenTelemetryIntegrationTest.java:
```diff
@@ -0,0 +1,74 @@
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.test.preview.microprofile.telemetry;
```
Perhaps don't use package org.wildfly.test.preview.

Using a package with 'preview' in it was probably a mistake in testsuite/preview/basic.

The general goal of stuff in WFP is that at some point it gets promoted to standard WildFly. With the vertx subsystem that could be very quickly. Promoting means moving the tests out of testsuite/preview so I think it's better to use a package structure that fits wherever the tests will end up.

bstansberry commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/preview/microprofile/telemetry/AbstractOpenTelemetryIntegrationTest.java:
```diff
@@ -0,0 +1,74 @@
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.test.shared.observability.setuptasks.ServiceNameSetupTask.SERVICE_NAME;
+
+public abstract class AbstractOpenTelemetryIntegrationTest {
```
@jasondlee Please review.

Review in general of course, but I put the comment here because soonish these classes will be promoted to standard WildFly and end up in testsuite/integration/microprofile. So now is the time to think whether this is a good name for a base class that will end up in that module, are the packages what you'll want etc. IOW let's plan for promotion.

jasondlee replied:
@gaol There's already a base otel test class in ts/integ/mp. To make promoting these tests out of preview, I'd suggest, say, forking and extending that class. It looks like the only new functionality you're adding is the log retrieval, so if all else is the same, there will be much less to merge/migrate when that time comes.

rhusar commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/preview/microprofile/telemetry/LoggingServerSetupTask.java:
```diff
@@ -0,0 +1,101 @@
+ *
+ * @author <a href="mailto:aoingl@gmail.com">Lin Gao</a>
+ */
+public class LoggingServerSetupTask implements ServerSetupTask {
```
Sorry but I don't see why this isn't just implementing `org.jboss.as.test.shared.ManagementServerSetupTask` instead which would significantly simplify this class and make it actually readable.

rhusar commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/preview/microprofile/telemetry/AbstractOpenTelemetryIntegrationTest.java:
```diff
@@ -0,0 +1,74 @@
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.test.shared.observability.setuptasks.ServiceNameSetupTask.SERVICE_NAME;
+
+public abstract class AbstractOpenTelemetryIntegrationTest {
```
I think it's the abstract class that should define `@DockerRequired` since it's the one defining a `@Testcontainer`. I believe these annotations inherited.

gaol commented:
Thank you everyone for the reviews, I will update this pr this week. :)

gaol committed:
Remove ts.copy-wildfly-standalone-embedded-broker exection

gaol committed:
[WFLY-19954] Add the wildfly-vertx-feature-pack to WildFly Preview

This is a follow up to add tests on the vertx instance usage in opentelemetry subsystem

gaol commented:
Updated this PR with:
* Added a separate commit to remove the `ts.copy-wildfly-standalone-embedded-broker` execution.
* Renamed the package to `org.wildfly.test.integration.observability.opentelemetry` to match the package name of tests in `testsuite/integration/microprofile`.
* Copied the abstract classes and REST resource classes from `testsuite/integration/microprofile` to make the tests easier to promote when the vertx feature pack gets promoted. The copied classes can just be removed at that time.
* Updated both `VertxSubsystemSetupTask` and `LoggingServerSetupTask` to extend `ManagementServerSetupTask` for easier set up and more readable.

jasondlee commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/integration/observability/opentelemetry/LoggingServerSetupTask.java:
```diff
@@ -0,0 +1,44 @@
+                .build());
+    }
+
+}
```
This looks MUCH better. +1 :)

rhusar replied:
Indeed - though there is still room for improvement :)

jasondlee commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/integration/observability/opentelemetry/OpenTelemetryIntegrationWithVertxTestCase.java:
```diff
@@ -0,0 +1,51 @@
+ * There will be a server log to indicate the usage.
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({OpenTelemetryWithCollectorSetupTask.class, ServiceNameSetupTask.class, OpenTelemetryIntegrationWithVertxTestCase.LoggingWithVertxServerSetupTask.class, VertxSubsystemSetupTask.class})
```
If you're not explicitly checking for a particular service name in a trace, the `ServiceNameSetupTask` is not needed here.

gaol replied:
OK, removed the usage of ServiceNameSetupTask, and use the name of the archive for the tests.

Actually checking for traces is not the main goal for this test case, it tries to send rest request to trigger the tracer so that the necessary messages to be checked will be in the log files.

jasondlee commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/integration/observability/opentelemetry/OpenTelemetryIntegrationWithoutVertxTestCase.java:
```diff
@@ -0,0 +1,43 @@
+ * There will be a server log to indicate a Vertx instance is created by smallerye-opentelemetry.
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({OpenTelemetryWithCollectorSetupTask.class, ServiceNameSetupTask.class, LoggingServerSetupTask.class})
```
Same here.

rhusar commented on testsuite/preview/expansion/src/test/java/org/wildfly/test/integration/observability/opentelemetry/LoggingServerSetupTask.java:
```diff
@@ -0,0 +1,44 @@
+        super(createContainerConfigurationBuilder()
+                .setupScript(createScriptBuilder()
+                        .startBatch()
+                        .add("/subsystem=logging/file-handler=otel-exporter:add(file={relative-to=jboss.server.log.dir, path=" + smallryeOpentelemetryLogFile + "}, append=false, level=DEBUG)")
```
FWIW the preferred way is to use the method variant with a pattern and %s expression with params, i.e. this one `org.jboss.as.test.shared.ManagementServerSetupTask.CommandSet#add(java.lang.String, java.lang.Object...)` but this works too...

gaol replied:
I am glad to know that, thank you :)

gaol commented:
Minor changes were done to address the feedback above, thank you guys :+1:

bstansberry commented:
Thanks @gaol and @jasondlee
